### PR TITLE
Adds more functionality to the Medical Kiosk with upgrades.

### PR DESCRIPTION
**Pull request by ArcaneMusic**
## About The Pull Request

So the feedback I got from players after adding the medical kiosk is that A) players will never build one of these considering where it's sitting on the tech tree, and B) Since it costs credits to use, it should at least be competitive with the advanced medical scanner in order to see use.

So to Fix point A, I'm going to be adding these to medical lobby on each map (However I'm currently struggling with mapmerger, bare with me).
For point B, I've added a T2 upgrade tier, which will provide radiation levels and tell players if the have any phobias, in addition to the T3 upgrades. In addition, I've also added some text coloring for brain damage and radiation, to help make irradiated or brain-damaged players better know they need treatment.
Kiosk Comparison
(Shows T1 vs T4 Upgrades)

## Why It's Good For The Game

Gives players more incentive to use the medical kiosk, as well as gives players an easier option mid-round to check their radiation, considering how prevalent radioactive contamination has been recently.

##To do list:

- [x] Add T2 Upgrades to the Medical Kiosk
- [x] Add Medical Kiosk to Medical Lobby on all maps.
- [x] Make the cost to use the machine adjustable.

## Changelog
:cl:
add: A medical kiosk now exists in medbay at shift start on all maps.
add: Medical kiosks now have a T2 Upgrade, showing brain traumas and radiation levels.
tweak: Players can now multitool the kiosk to change the cost to get scanned, up to ludicrous levels.
/:cl:

### CODE REVIEW

**Comment by ExcessiveUseOfCobblestone** [comment]
need to make the analyzers less available :smirk:

**Comment by MonsWTF** [comment]
Out of curiosity...are doctors able to adjust the amount charged?

**Comment by ArcaneMusic** [comment]
> Out of curiosity...are doctors able to adjust the amount charged?

Not currently, but once I have the mapping done I could make that adjustable with a multitool on the board.

**Comment by Shapsy** [comment]
You could also add another tier of health scanners for MD's that retain most of the info they show now, and put limits on the scanners found around the station. Could be they only tell you the name of a disease rather than the transmission method/stage/cure, only tell you detected chems over a certain concentration like 5u/don't show amounts, etc etc

That way if you want an in depth diagnosis, you'd have to find a doctor or pay to use a scanner

**Comment by ArcaneMusic** [comment]
Alright, the medical kiosk can now be adjusted by using a multitool on the board to set a custom price, and the price is reflected on both examine and dynamically within the UI.
CustomPrice

**Comment by Tlaltecuhtli** [comment]
make the tier 4 give you the IQ of the patient ( randomly generated)

File: code/game/machinery/medical_kiosk.dm [hunk 1]
```diff
@@ -9,31 +9,34 @@
 	density = TRUE
 	circuit = /obj/item/circuitboard/machine/medical_kiosk
 	payment_department = ACCOUNT_MED
-	var/default_price = 5  //I'm defaulting to a low price on this, but in the future I wouldn't have an issue making it more or less expensive.
-	var/scan_active = null  //Shows if the machine is being used. resets upon new viewer.
-	var/adv_scan_active = null  //Shows if the machine has upgraded functionality
+	var/default_price = 5           //I'm defaulting to a low price on this, but in the future I wouldn't have an issue making it more or less expensive.
+	var/active_price = 5            //Change by using a multitool on the board.
+	var/scan_active = null          //Shows if the machine is being used. resets upon new viewer.
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/machinery/medical_kiosk.dm]:
```suggestion
	var/scan_active         //Shows if the machine is being used. resets upon new viewer.
```

File: code/game/machinery/medical_kiosk.dm [hunk 2]
```diff
@@ -9,31 +9,34 @@
...
+	var/upgrade_scan_active = null
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/machinery/medical_kiosk.dm]:
```suggestion
	var/upgrade_scan_active
```

File: code/game/machinery/medical_kiosk.dm [hunk 3]
```diff
@@ -9,31 +9,34 @@
...
+	var/adv_scan_active = null      //Shows if the machine has upgraded functionality
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/machinery/medical_kiosk.dm]:
```suggestion
	var/adv_scan_active //Shows if the machine has upgraded functionality
```

Comment (ExcessiveUseOfCobblestone) [inline code/game/machinery/medical_kiosk.dm]: are these boos >:(

Comment (ArcaneMusic) [reply code/game/machinery/medical_kiosk.dm]: Good coding style is not something I'm great at, MB

**Comment by Time-Green** [comment]
You could also add genetic stability if you’re looking for more tier stuff

File: code/game/machinery/medical_kiosk.dm (hunk 3 above)
Comment (ExcessiveUseOfCobblestone) [reply code/game/machinery/medical_kiosk.dm]: boos need to use `TRUE`/`FALSE` defines and should have default values of either (in this case they'd be `FALSE`)

**Comment by ArcaneMusic** [comment]
Alright, with some help I got the variables working as booleans (interacting with JS "booleans"), and not just cutting corners with 0s and 1s.

**Comment by ExcessiveUseOfCobblestone** [comment]
is this ready2go

**Comment by ArcaneMusic** [comment]
> is this ready2go

Yup, locked and loaded.

File: tgstation.dme [hunk 4]
```diff
@@ -351,6 +351,7 @@
 #include "code\datums\brain_damage\split_personality.dm"
 #include "code\datums\components\_component.dm"
 #include "code\datums\components\anti_magic.dm"
+#include "code\datums\components\archaeology.dm"
```
Comment (ExcessiveUseOfCobblestone) [inline tgstation.dme]: 😏

Comment (ArcaneMusic) [reply tgstation.dme]: 😓

Comment (ArcaneMusic) [reply tgstation.dme]: Yikes, nice catch.

**Pull request merged**